src/runtime_src/tools/xclbinutil/XclBinUtilMain.cxx

```diff
@@ -584,7 +586,22 @@ int main_(int argc, const char** argv) {
       oInfoFile.close();
     }
   }
+  
+  if(fileCheck){
+    std::string fileCheckText = "";
 
+    if(!xclBin.checkForPlatformVbnv()){
+         fileCheckText += "platformVBNV information";
+    }
+    if(!xclBin.checkForValidSection()){
+      if (!fileCheckText.empty())
+        fileCheckText += " and ";
+      fileCheckText += "at least one section";
+    }
+    if (!fileCheckText.empty()) {
+      throw std::runtime_error("ERROR: The xclbin is missing " + fileCheckText + " required by the 'file' command to identify its file type and display file characteristics");
```

Comment:
Move this check before line 596 to simplify the logic, e.g. use early-exit / early-check.